central/policycategoryedge/datastore/datastore_impl.go

```diff
@@ -144,24 +147,22 @@ func (ds *datastoreImpl) DeleteByQuery(ctx context.Context, q *v1.Query) error {
 	return ds.storage.DeleteByQuery(ctx, q)
 }
 
-func convertMany(edges []*storage.PolicyCategoryEdge, results []searchPkg.Result) ([]*v1.SearchResult, error) {
-	if len(edges) != len(results) {
-		return nil, errors.Errorf("expected %d results, got %d", len(edges), len(results))
-	}
+type PolicyCategoryEdgeSearchResultConverter struct{}
 
-	outputResults := make([]*v1.SearchResult, len(edges))
-	for index, edge := range edges {
-		outputResults[index] = convertOne(edge, &results[index])
-	}
-	return outputResults, nil
+func (c *PolicyCategoryEdgeSearchResultConverter) BuildName(result *searchPkg.Result) string {
+	// Name is already populated from ID
```

Comment:
Why would you choose to populate Name outside of this result converter? This converter is not complete if we do the `Name = ID` in the search function.

Reply:
@c-du  I can do a follow up PR as a cleanup. Wont be too difficult to do for name